plugin/src/test/java/io/jenkins/plugins/casc/SchemaGenerationTest.java

```diff
@@ -0,0 +1,144 @@
+package io.jenkins.plugins.casc;
+
+import hudson.DescriptorExtensionList;
+import hudson.model.UpdateSite;
+import hudson.tasks.Maven;
+import hudson.util.DirScanner;
+import io.jenkins.plugins.casc.core.*;
+import io.jenkins.plugins.casc.impl.configurators.DataBoundConfigurator;
+import io.jenkins.plugins.casc.impl.configurators.GlobalConfigurationCategoryConfigurator;
+import io.jenkins.plugins.casc.impl.configurators.HeteroDescribableConfigurator;
+import io.jenkins.plugins.casc.misc.JenkinsConfiguredWithCodeRule;
+import org.junit.Rule;
+import org.junit.Test;
+import java.util.*;
+
+
+import java.util.LinkedHashSet;
+
+
+public class SchemaGenerationTest{
+
+
+    @Rule
+    public JenkinsConfiguredWithCodeRule j = new JenkinsConfiguredWithCodeRule();
+
+    @Test
+    public void schemaShouldSucceed() {
+
+
+        /**
+         * Used to generate the schema for root configurators
+         */
+        LinkedHashSet linkedHashSet = new LinkedHashSet<>(ConfigurationAsCode.get().getRootConfigurators());
```

Comment:
You should be able to move this whole thing into `generateSchema` and have a test call `generateSchema` without issue. Since your tests will provide the Jenkins Instance because of ```
@Rule
public JenkinsConfiguredWithCodeRule j = new JenkinsConfiguredWithCodeRule();
```